include/EASTL/bonus/ring_buffer.h

```diff
@@ -1505,7 +1505,12 @@ namespace eastl
 	template <typename T, typename Container, typename Allocator>
 	inline bool operator==(const ring_buffer<T, Container, Allocator>& a, const ring_buffer<T, Container, Allocator>& b)
 	{
-		return (a.size() == b.size()) && (a.c == b.c);
+		const typename ring_buffer<T, Container, Allocator>::size_type sizeA = a.size();
```

Comment:
This is a minor thing but given that I'm requesting other changes. This is probably a good place to use `const auto` .

Reply:
as previously mentioned